build/shade/_dotnet-test.shade

```diff
@@ -17,17 +17,27 @@ configuration=''
 default configuration = 'Debug'
 
 @{
-    var projectFolder = Path.GetDirectoryName(projectFile);
-    var projectName = Path.GetFileName(projectFolder);
-
-    var noParallelTestProjects = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
-    if (!string.IsNullOrEmpty(NO_PARALLEL_TEST_PROJECTS))
+    var content = File.ReadAllText(projectFile);
+    // Desktop tests only work on Windows. Only run the test on xplat if it has a netcoreapp1.0 TFM.
+    if (content.Contains("\"testRunner\"") && (!IsLinux || content.Contains("netcoreapp1.0")))
     {
-        noParallelTestProjects.UnionWith(NO_PARALLEL_TEST_PROJECTS.Split((char)','));
+        var projectFolder = Path.GetDirectoryName(projectFile);
+        var projectName = Path.GetFileName(projectFolder);
+
+        var noParallelTestProjects = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+        if (!string.IsNullOrEmpty(NO_PARALLEL_TEST_PROJECTS))
+        {
+            noParallelTestProjects.UnionWith(NO_PARALLEL_TEST_PROJECTS.Split((char)','));
+        }
+
+        var testArgs = " --configuration " + configuration;
+        if (IsLinux)
+        {
+            testArgs += " -f netcoreapp1.0 -parallel none";
```

Comment:
IIRC it was because of async issues on just mono. There's a comment linking to issue https://github.com/xunit/xunit/issues/158. But issue has been resolved. Probably best to just try and see if linux testing passes with parallelization on.